Alignment/OfflineValidation/bin/PVtrends.cc

```diff
@@ -0,0 +1,205 @@
+#include <cstdlib>
+#include <string>
+#include <tuple>
+#include <iostream>
+#include <numeric>
+#include <functional>
+#include <unistd.h>
+
+#include "TFile.h"
+#include "TGraph.h"
+#include "TH1.h"
+
+#include "exceptions.h"
+#include "toolbox.h"
+#include "Options.h"
+
+#include "boost/filesystem.hpp"
+#include "boost/algorithm/string.hpp"
+#include "boost/container/vector.hpp"
+#include "boost/property_tree/ptree.hpp"
+#include "boost/property_tree/json_parser.hpp"
+#include "boost/optional.hpp"
+#include "fmt/printf.h"
+
+#include "TString.h"
+#include "TColor.h"
+
+#include "Alignment/OfflineValidation/interface/PreparePVTrends.h"
+#include "Alignment/OfflineValidation/interface/Trend.h"
+
+using namespace std;
+using namespace AllInOneConfig;
+namespace fs = boost::filesystem;
+namespace bc = boost::container;
+
+static const char *bold = "\e[1m", *normal = "\e[0m";
+static const float defaultConvertScale = 1000.;
+namespace pt = boost::property_tree;
+
+int trends(int argc, char *argv[]) {
+  // parse the command line
+
+  Options options;
+  options.helper(argc, argv);
+  options.parser(argc, argv);
+
+  //Read in AllInOne json config
+  pt::ptree main_tree;
+  pt::read_json(options.config, main_tree);
+
+  pt::ptree alignments = main_tree.get_child("alignments");
+  pt::ptree validation = main_tree.get_child("validation");
+  pt::ptree style = main_tree.get_child("style");
+
+  //Read all configure variables and set default for missing keys
+  string outputdir = main_tree.get<string>("output");
+  bool doRMS = validation.count("doRMS") ? validation.get<bool>("doRMS") : true;
+  bool doUnitTest = validation.count("doUnitTest") ? validation.get<bool>("doUnitTest") : false;
+  int nWorkers = validation.count("nWorkers") ? validation.get<int>("nWorkers") : 20;
+  TString lumiInputFile = style.count("lumiInputFile") ? style.get<string>("lumiInputFile")
+                                                       : "Alignment/OfflineValidation/data/lumiPerRun_Run2.txt";
+  fs::path lumiFile = lumiInputFile.Data();
+  fs::path pathToLumiFile = "";
+  if (!fs::exists(lumiFile)) {
+    bc::vector<fs::path> possible_base_paths;
+    boost::split(possible_base_paths, std::getenv("CMSSW_SEARCH_PATH"), boost::is_any_of(":"));
+    for (const fs::path &path : possible_base_paths) {
+      if (fs::exists(path / lumiFile)) {
+        pathToLumiFile = path / lumiFile;
+        break;
+      }
```

Comment:
why not just use [FileInPath](https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/interface/FileInPath.h ) which  already does this?